src/select/index.tsx

```diff
@@ -1,487 +1,229 @@
-	/** Custom aria attributes */
-	aria?: { [key: string]: string | null };
-	/** Set the disabled property of the control */
+import bundle from './select.nls';
+
+interface SelectProperties {
+	/** Callback called when user selects a value */
+	onValue(value: string): void;
+	/** The initial selected value */
+	initialValue?: string;
+	/** Options to display within the menu */
+	options: MenuOption[];
+	/** Property to determine how many items to render. Defaults to 6 */
+	itemsInView?: number;
+	/** Custom renderer for item contents */
+	itemRenderer?(properties: ItemRendererProperties): RenderResult;
+	/** placement of the select menu; 'above' or 'below' */
+	position?: PopupPosition;
+	/** Placeholder value to show when nothing has been selected */
+	placeholder?: string;
+	/** Property to determine if the input is disabled */
 	disabled?: boolean;
-	/** Function that accepts an option's data and index and returns a boolean */
-	getOptionDisabled?(option: T, index: number): boolean;
-	/** Function that accepts an option's data and index and returns a string id */
-	getOptionId?(option: T, index: number): string;
-	/** Function that accepts an option's data and returns a DNode label */
-	getOptionLabel?(option: T): DNode;
-	/** Function that accepts an option's data and index and returns a boolean */
-	getOptionSelected?(option: T, index: number): boolean;
-	/** Function that accepts an option's data and returns a string, used for matching an option on keydown */
-	getOptionText?(option: T): string;
-	/** Function that accepts an option's data and index and returns a string value */
-	getOptionValue?(option: T, index: number): string;
-	/** Renders helper text with the Select */
+	/** Sets the helper text of the input */
 	helperText?: string;
-	/** Adds a <label> element with the supplied text */
-	label?: string;
-	/** Hides the label from view while still remaining accessible for screen readers */
-	labelHidden?: boolean;
-	/** The name of the select field */
-	name?: string;
-	/** Handler for when the element is blurred */
-	onBlur?(): void;
-	/** Handler for when the element is focused */
-	onFocus?(): void;
-	/** Handler for when the pointer moves out of the element */
-	onOut?(): void;
-	/** Handler for when the pointer moves over the element */
-	onOver?(): void;
-	/** Handler for when the value of the widget changes */
-	onValue?(option: T): void;
-	/** Array of any type of data for the options */
-	options?: T[];
-	/** Optional placeholder text, only valid for custom select widgets (useNativeElement must be false or undefined) */
-	placeholder?: string;
-	/** Makes the select readonly (it may be focused but not changed) */
-	readOnly?: boolean;
-	/** Sets the input as required to complete the form */
+	/** The label to show */
+	label?: RenderResult;
+	/** Boolean to indicate if field is required */
 	required?: boolean;
-	/** Use the native <select> element if true */
-	useNativeElement?: boolean;
```

Comment:
Is there no longer going to be an option to use the native element?

Reply:
I think that it is better that we have a separate native select widget.